metricbeat/helper/prometheus/prometheus.go

```diff
@@ -139,11 +141,7 @@ type MetricsMapping struct {
 	ExtraFields map[string]string
 }
 
-func (p *prometheus) GetProcessedMetrics(mapping *MetricsMapping) ([]common.MapStr, error) {
-	families, err := p.GetFamilies()
-	if err != nil {
-		return nil, err
-	}
+func (p *prometheus) processedMetrics(families []*dto.MetricFamily, mapping *MetricsMapping) ([]common.MapStr, error) {
```

Comment:
Nit. Expose this method directly as `func (p *prometheus) ProcessMetrics(families []*dto.MetricFamily, mapping *MetricsMapping) ([]common.MapStr, error)`, and remove `GetSharedProcessedMetrics`. This helper library doesn't need to know anything about "Shared" data.